Added marker support

judax opened the pull request.

judax committed:
Removed some logs.

Change-Id: I9230381cd09a11d3c471949d46a47086bc3cda5d

lincolnfrog commented on android_webview/test/shell/tango/jni/TangoHandler.h:
```diff
@@ -68,6 +77,54 @@ class TangoHandlerEventListener
 };
 
+class Marker
+{
```
Should do a pass moving {s to the previous line on the entire PR.

lincolnfrog commented on android_webview/test/shell/tango/jni/TangoHandler.h:
```diff
@@ -68,6 +77,54 @@ class TangoHandlerEventListener
+public:
+  Marker(TangoMarkers_MarkerType type, int id, const std::string& content, const double* position, const double* orientation, const float* modelMatrix): type(type), id(id), content(content)
+  {
+    memcpy(this->position, position, sizeof(this->position));
```
Why have all these redundant properties? Can we just have a matrix (more consistent with the other APIs)?

judax replied:
I am ok dropping the orientation and position. This has been a legacy thing as in the initial marker detection I provided these values. We will have to remember to update the marker example that now it should be in the not only in the tango folder (although ARKit does not support markers yet).

lincolnfrog commented on android_webview/test/shell/tango/jni/TangoHandler.h:
```diff
@@ -68,6 +77,54 @@ class TangoHandlerEventListener
+  std::string content;
+  double position[3];
+  double orientation[4];
+  float modelMatrix[16];
```
something I noticed when working on ARCore is that the Google c-style uses underscore_case, not camelCase for its variable names. I don't think we should necessarily change all our code since there is a lot of it (and maybe they use camelCase in Chromium?). Anyway, just thought I would mention this - no action required.

judax replied:
Good point. I do not fully know if that is the case for Chromium that it seems that is using a pretty inconsistent way of handling style. Some functions use camel case with the initial character being in capital letters, some private members use m_... it seems it is consistent in the layer though (IDL Vs Device). I am ok making the change we decide upon for our code like TangoHandler, AnchorManager, etc...

lincolnfrog commented on android_webview/test/shell/tango/jni/TangoHandler.h:
```diff
@@ -82,12 +139,12 @@ class TangoHandler {
 
-  void onCreate(JNIEnv* env, jobject activity, 
-                int activityOrientation, int sensorOrientation);
+  void onCreate(JNIEnv* env, jobject activity, int activityOrientation, int sensorOrientation);
```
it seems like this line is longer than 80 characters to me...

lincolnfrog commented on device/vr/android/tango/tango_vr_device.cc:
```diff
@@ -306,6 +310,61 @@ void TangoVRDevice::RemoveAnchor(uint32_t identifier) {
+        return mojomMarkers;
+    }
+    std::vector<Marker> markers;
+    if (TangoHandler::getInstance()->getMarkers(mt, markerSize, markers))
```
Doesn't this return an enum? Explicit check?

judax replied:
No, it returns a boolean indicating if the marker detection was correct (even if no marker were detected).

lincolnfrog commented on third_party/tango/libtango_chromium/TangoHandler.h:
```diff
@@ -68,6 +77,54 @@ class TangoHandlerEventListener
+class Marker
+{
+public:
+  Marker(TangoMarkers_MarkerType type, int id, const std::string& content, const double* position, const double* orientation, const float* modelMatrix): type(type), id(id), content(content)
```
this line is way too long

lincolnfrog commented on third_party/tango/libtango_markers/tango_markers.h:
```diff
@@ -0,0 +1,134 @@
+  /// TangoMarkers_detectMarkers() function.
+  /// The marker pose defines a marker local frame, in which:
+  ///  X = to the right on the tag
+  ///  Y = to the up on the tag
```
this isn't worded correctly. I would change both of these comments for X and Y.  Maybe just say the tag's up vector and the tag's right vector?

judax replied:
This is not our code but the marker detection library header.

lincolnfrog commented on third_party/tango/libtango_markers/tango_markers.h:
```diff
@@ -0,0 +1,134 @@
+  /// The marker pose defines a marker local frame, in which:
+  ///  X = to the right on the tag
+  ///  Y = to the up on the tag
+  ///  Z = pointing out of the tag towards the user.
```
this comment makes more sense

lincolnfrog commented on third_party/tango/libtango_markers/tango_markers.h:
```diff
@@ -0,0 +1,134 @@
+
+/// @brief Detect one or more markers in the input image.
+/// @param image_buffer The image buffer. Cannot be NULL.
+/// @param camera_id The identification of the camera that captured the
```
I would say "The id of the camera" instead of "identification" here

lincolnfrog reviewed (approved):
Mostly stylistic comments. Looks good!

judax committed:
Markers now only return the modelMatrix (not the position and
orientation)
so the API is more consistent with VRHits. Commented out some logs.
Fixed some syntax issues according to the PR review.

Change-Id: I8698c522000803ce38d0a826581c585bbfb32108

judax commented:
@lincolnfrog I made the suggested changes. Please, review it whenever you can.

judax committed:
Fixed crash because the listener was not being removed.

Change-Id: Ib3087e64c574cfaa2df7cdd0468aeabc6e6296bb

judax committed:
Removed some unused APKs.

Change-Id: I4ca889361732d4cb6650a241be65eb335cd23978

judax committed:
Fixed a mistake in the README regarding the ninja command.

Change-Id: I42bcb4ba4b03e8fd7a819e636c3b0518931d63cb

lincolnfrog reviewed (approved):
Still lots of { on the next line, but we need to get this in so I am approving.

judax commented:
Tried to fix as many as I could. My find and replace regex might not have been bulletproof :)
